doc/common-conventions.rst

```diff
@@ -0,0 +1,61 @@
+Common Conventions
+==================
+
+Requests made via portal interfaces generally involve user interaction, and
+dialogs that can stay open for a long time. Therefore portal APIs don't just use
+async method calls (which time out after at most 25 seconds), but instead return
+results via a Response signal on Request objects.
+
+Portal APIs don't use properties very much. This is partially because we need to
+be careful about the flow of information, and partially because it would be
+unexpected to have a dialog appear when you just set a property.
```

Comment:
Maybe add "However, every portal has at least one ``version`` property that specifies the maximum version supported by the xdg-desktop-portal.".